app/Writer/UserWriter.php

```diff
@@ -2,36 +2,37 @@
 
 namespace App\Writer;
 
-use App\Event\NewUserSaved;
+use App\Event\NewUserSavedEvent;
 use App\Model\Contract\User;
-use App\Repository\UserRepository;
-use Illuminate\Events\Dispatcher;
+use App\Repository\Contract\UserRepository;
 
 class UserWriter
 {
     /** @var UserRepository */
     private $userRepository;
 
-    /** @var Dispatcher */
+    /** @var \App\Event\Dispatcher */
     private $eventDispatcher;
 
     /**
-     * @param UserRepository $userRepository
-     * @param Dispatcher     $eventDispatcher
+     * @param UserRepository        $userRepository
+     * @param \App\Event\Dispatcher $eventDispatcher
      */
-    public function __construct(UserRepository $userRepository, Dispatcher $eventDispatcher)
+    public function __construct(UserRepository $userRepository, \App\Event\Dispatcher $eventDispatcher)
     {
         $this->userRepository  = $userRepository;
         $this->eventDispatcher = $eventDispatcher;
     }
 
     /**
      * @param User $user
+     *
+     * @return bool
      */
     public function addUser(User $user)
     {
         $result = $this->saveUser($user);
-        $event  = new NewUserSaved($user);
+        $event  = new NewUserSavedEvent($user);
 
         $this->eventDispatcher->dispatch($event,[$user]);
```

Comment:
**Suggestion:** The `addUser` method in the `UserWriter` class creates a new `NewUserSavedEvent` but does not check if the user was saved successfully before dispatching the event. Consider checking the result of the `saveUser` method before dispatching the event. [enhancement]
```suggestion
    public function addUser(User $user)
    {
        $result = $this->saveUser($user);
        if ($result) {
            $event  = new NewUserSavedEvent($user);
            $this->eventDispatcher->dispatch($event,[$user]);
        }
        return $result;
    }
```